### PR TITLE
NAS 100964

### PR DESCRIPTION
**Pull request by lola27**

### CODE REVIEW

File: src/app/pages/services/services.component.ts [hunk 1]
```diff
@@ -129,12 +127,24 @@ export class Services implements OnInit {
...
-      confirm.subscribe(res => {
-        if (res) {
-          this.updateService(rpc, service);
-        }
-      })
+      if (service.title == 'iscsitarget') {
+        this.iscsiService.getGlobalSessions().subscribe(
+          (res) => {
+            const msg = res.length == 0 ? '' : T('<font color="red"> You have ') + res.length + T(' pending active iSCSI connection(s).</font><br>Stop the service?');
+            this.dialog.confirm(T('Alert'),  msg == '' ? T('Stop this service?') : msg, true, T('Stop')).subscribe(dialogRes => {
```
Comment (Mrt134) [inline src/app/pages/services/services.component.ts]: "Stop this service?"
This is fine as-is, but would it be more helpful for the message to show which service will be stopped? "Stop iSCSI?", "Stop Dynamic DNS?", "Stop `service-name`?"

File: src/app/pages/sharing/iscsi/associated-target/associated-target-list/associated-target-list.component.ts [hunk 2]
```diff
@@ -59,4 +67,14 @@ export class AssociatedTargetListComponent {
       });
     });
   }
+
+  warningMsg(item) {
+    let warningMsg = '<font color="red">';
+    for (let i = 0; i < this.iscsiSessions.length; i++) {
+      if (this.iscsiSessions[i].target.split(':')[1] == item.target) {
+        warningMsg += T('Warnning: Target in use');
```
Comment (Mrt134) [inline src/app/pages/sharing/iscsi/associated-target/associated-target-list/associated-target-list.component.ts]:
```suggestion
        warningMsg += T('Warning: iSCSI Target is already in use.');
```

File: src/app/pages/services/services.component.ts [hunk 3]
```diff
@@ -129,12 +127,24 @@ export class Services implements OnInit {
...
+              if (dialogRes) {
+                this.updateService(rpc, service);
+              }
+            });
+          }
+        )
+      } else {
+        this.dialog.confirm(T('Alert'), T('Stop this service?'), true, T('Stop')).subscribe(res => {
```
Comment (Mrt134) [inline src/app/pages/services/services.component.ts]: Same comment as above

File: src/app/pages/sharing/iscsi/target/target-list/target-list.component.ts [hunk 4]
```diff
@@ -24,7 +28,7 @@ export class TargetListComponent {
       name : 'Target Alias',
       prop : 'alias',
     },
-  ];
+  ];6
```
Comment (Mrt134) [inline src/app/pages/sharing/iscsi/target/target-list/target-list.component.ts]: Is this a typo?

File: src/app/pages/services/services.component.ts [hunk 5]
```diff
@@ -129,12 +127,24 @@ export class Services implements OnInit {
...
-      let confirm = this.confirmService.confirm(T('Alert'), T('Stop this service?'), T('Stop'));
-      confirm.subscribe(res => {
-        if (res) {
-          this.updateService(rpc, service);
-        }
-      })
+      if (service.title == 'iscsitarget') {
+        this.iscsiService.getGlobalSessions().subscribe(
+          (res) => {
+            const msg = res.length == 0 ? '' : T('<font color="red"> You have ') + res.length + T(' pending active iSCSI connection(s).</font><br>Stop the service?');
```
[the remark this answers is not in the document]
Comment (lola27) [reply src/app/pages/services/services.component.ts]: @Mrt134 confirmed with William. It's ok to remove 'pending'. I pushed a commit that will make the change.

**Pull request merged**